tests/LogFormatTest.php

```
@@ -31,4 +31,25 @@ public function testTryFromUnknownValueReturnsNull(): void
+        $this->assertSame(LogFormat::json(), LogFormat::tryFrom('json'));
+        $this->assertNotSame(LogFormat::text(), LogFormat::json());
+    }
+
+    public function testInstancesRejectMutation(): void
+    {
+        // Singletons must not be mutable, otherwise a write through one
+        // reference would corrupt every cached caller.
+        $format = LogFormat::text();
+
+        $this->expectException(\LogicException::class);
+        $format->value = 'mutated';
```

PHPStan (2.1.51)

[error] Property Timewave\Logger\Enums\LogFormat::$value is not writable.


(assign.propertyReadOnly)